cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -621,6 +616,54 @@ mkConsensusTracers mbEKGDirect trSel verb tr nodeKern fStats = do
      when (isRollForward ev) $
        sendEKGDirectCounter ekgDirect "cardano.node.metrics.served.header.counter.int"
 
+traceBlockFetchServerMetrics
+  :: forall blk. ()
+  => Maybe EKGDirect
+  -> STM.TVar Int64
+  -> STM.TVar Int64
+  -> STM.TVar SlotNo
+  -> Tracer IO (TraceBlockFetchServerEvent blk)
+  -> Tracer IO (TraceBlockFetchServerEvent blk)
+traceBlockFetchServerMetrics Nothing _ _ _ tracer = tracer
+traceBlockFetchServerMetrics (Just ekgDirect) tBlocksServed tLocalUp tMaxSlotNo tracer = Tracer bsTracer
+
+  where
+    bsTracer :: TraceBlockFetchServerEvent blk -> IO ()
```

Comment:
This adds the local upstreaminess metric which tries to answer the question: how many times we serve a block which slot is either equal or ahead of last served block.

I would like to see a description how this related to the upstreaminess metric that is used to choose peers.

Reply:
I don't think we should mention that at all. This is a useful metric on its own, regardless of if that is how down stream peers may judge the node in the future.